src/finite_elasticity_routines.f90

```diff
@@ -3448,11 +3466,740 @@ SUBROUTINE FINITE_ELASTICITY_GAUSS_CAUCHY_TENSOR(EQUATIONS_SET,DEPENDENT_INTERPO
+              & C(i)*EVALUES(3)**(C(i+3)/2.0_DP-1.0_DP)*TEMP3
+          END DO
+          SLOPE=TEMP(1,1)*C(7)
+          lambda_a=lambda_a-VALUE/SLOPE
+          !IF (lambda_a.LE.0.0_DP) THEN
+          ! lambda_a=0.1_DP
+          !END IF
+          !lambda_a=lambda_a-0.001
+
+          F_a_inv=0.0_DP
+          F_a_inv(1,1)=1.0_DP/lambda_a
+          F_a_inv(2,2)=1.0_DP
+          F_a_inv(3,3)=1.0_DP
+
+          CALL MATRIX_PRODUCT(DZDNU,F_a_inv,F_e,ERR,ERROR,*999)
+          CALL MATRIX_TRANSPOSE(F_e,F_e_T,ERR,ERROR,*999)
+          CALL MATRIX_PRODUCT(F_e_T,F_e,C_e,ERR,ERROR,*999)
+
+          CALL DSYEV('V','U',3,C_e,3,EVALUES,WORK,-1,ERR)
+          IF(ERR.NE.0) WRITE(*,*) "Error in Eigenvalue computation"
+          LWORK=MIN(LWMAX,INT(WORK(1)))
+          CALL DSYEV('V','U',3,C_e,3,EVALUES,WORK,LWORK,ERR)
+          IF(ERR.NE.0) WRITE(*,*) "Error in Eigenvalue computation"
+          EVECTOR_1=C_e(:,1)
+          EVECTOR_2=C_e(:,2)
+          EVECTOR_3=C_e(:,3)
+
+          DO i=1,3
+            DO j=1,3
+              EMATRIX_1(i,j)=EVECTOR_1(i)*EVECTOR_1(j)
+              EMATRIX_2(i,j)=EVECTOR_2(i)*EVECTOR_2(j)
+              EMATRIX_3(i,j)=EVECTOR_3(i)*EVECTOR_3(j)
+            END DO
+          END DO
+
+          CALL MATRIX_PRODUCT(F_a_inv,EMATRIX_1,N1,ERR,ERROR,*999)
+          CALL MATRIX_PRODUCT(N1,F_a_inv,N1,ERR,ERROR,*999) ! F_a_inv=F_a_inv_T
+          CALL MATRIX_PRODUCT(F_a_inv,EMATRIX_2,N2,ERR,ERROR,*999)
+          CALL MATRIX_PRODUCT(N2,F_a_inv,N2,ERR,ERROR,*999) ! F_a_inv=F_a_inv_T
+          CALL MATRIX_PRODUCT(F_a_inv,EMATRIX_3,N3,ERR,ERROR,*999)
+          CALL MATRIX_PRODUCT(N3,F_a_inv,N3,ERR,ERROR,*999) ! F_a_inv=F_a_inv_T 
+
+          FREE_ENERGY=0.0_DP
+          DO i=1,3
+            FREE_ENERGY=FREE_ENERGY+C(i)/C(i+3)*( &
+              & EVALUES(1)**(C(i+3)/2.0_DP)+ &
+              & EVALUES(2)**(C(i+3)/2.0_DP)+ &
+              & EVALUES(3)**(C(i+3)/2.0_DP)-3.0_DP)
+          END DO
+          FREE_ENERGY=C(7)*FREE_ENERGY
+
+          VALUE=XB_ENERGY_PER_VOLUME-(FREE_ENERGY-FREE_ENERGY_0)
+        ENDIF
+      ENDDO
+    
+      PIOLA_TENSOR=0.0_DP
+      DO i=1,3
+        PIOLA_TENSOR=PIOLA_TENSOR+ &
+          & C(i)*EVALUES(1)**(C(i+3)/2.0_DP-1.0_DP)*N1+ &
+          & C(i)*EVALUES(2)**(C(i+3)/2.0_DP-1.0_DP)*N2+ &
+          & C(i)*EVALUES(3)**(C(i+3)/2.0_DP-1.0_DP)*N3
+      END DO
+      PIOLA_TENSOR=PIOLA_TENSOR*C(7)+2.0_DP*P*AZU
+      
+      !store lambda_f, so it can be used in the CellML file
+      NULLIFY(FIELD_VARIABLE)
+      CALL FIELD_VARIABLE_GET(DEPENDENT_FIELD,FIELD_U1_VARIABLE_TYPE,FIELD_VARIABLE,ERR,ERROR,*999)
+      dof_idx=FIELD_VARIABLE%COMPONENTS(1)%PARAM_TO_DOF_MAP%GAUSS_POINT_PARAM2DOF_MAP%GAUSS_POINTS(GAUSS_POINT_NUMBER, &
+        & ELEMENT_NUMBER)
+      lambda_f=SQRT(AZL(1,1))
+      CALL DISTRIBUTED_VECTOR_VALUES_SET(FIELD_VARIABLE%PARAMETER_SETS%SET_TYPE(FIELD_VALUES_SET_TYPE)%PTR%PARAMETERS,dof_idx, &
+        & lambda_f,ERR,ERROR,*999)
```

Comment:
Should really be using the field values set routines here rather than directly accessing the underlying distributed vector.